src/services/index/generate-submain-sitemap-index/generate-submain-sitemap-index-service.ts

```
@@ -0,0 +1,22 @@
+import * as sm from "sitemap";
+import { XmlRequest } from "../../../model/index/common/XmlRequest";
+import { HOSTNAME } from "../constants";
+import { formUrls } from "./generate-urls";
+
+export function generateSubmainSitemapIndexService(
+  queryStringParameters: XmlRequest,
+): Promise<string> {
+  return new Promise(async (resolve) => {
+    const sitemap = sm.createSitemap({
+      hostname: HOSTNAME,
+      urls: await formUrls(queryStringParameters.number),
+    });
+
+    sitemap.toXML((err, xml) => {
+      if (err) {
```

Error is hidden. Need to log or throw error with details.